packages/react-magma-dom/src/components/Dropdown/DropdownExpandableMenuButton.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { AccordionButton, AccordionButtonProps } from '../Accordion';
+import { IconWrapper, menuBackground } from './DropdownMenuItem';
+import { IconProps } from 'react-magma-icons';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { DropdownContext } from './Dropdown';
+import { DropdownExpandableContext } from './DropdownExpandableMenuGroup';
+import { useForkedRef } from '../../utils';
+
+export interface DropdownExpandableMenuButtonProps
+  extends AccordionButtonProps {
+  icon?: React.ReactElement<IconProps>;
+  testId?: string;
+}
+
+const StyledAccordionButton = styled(AccordionButton)<{
+  hasIcon?: boolean;
+  icon?: React.ReactElement<IconProps>;
+}>`
+  font-weight: 400;
+  padding: ${props =>
+    !props.icon && props.hasIcon
+      ? `${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing05} ${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing11}`
+      : `${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing05}`};
+  margin: 0;
+  border-top: 0;
+  &:hover,
+  &:focus {
+    background: ${menuBackground};
+  }
+  > span {
+    display: flex;
+  }
+`;
+
+const StyledIconWrapper = styled(IconWrapper)`
+  justify-content: center;
+  /* align-items: center; */
+`;
+
+export const DropdownExpandableMenuButton = React.forwardRef<
+  HTMLDivElement,
+  DropdownExpandableMenuButtonProps
+>((props, forwardedRef) => {
+  const { children, icon, testId, ...other } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const context = React.useContext(DropdownContext);
+  const expandableContext = React.useContext(DropdownExpandableContext);
+
+  const ownRef = React.useRef<HTMLDivElement>();
+  const ref = useForkedRef(forwardedRef, ownRef);
+
+  React.useEffect(() => {
+    context.registerDropdownMenuItem(context.itemRefArray, ownRef);
+  }, []);
+
+  return (
+    <StyledAccordionButton
+      {...other}
+      ref={ref}
+      icon={props.icon}
```

Comment:
Since you already declared icon above (line 46) you can just use `icon={icon}` here OR `icon`